kamaki/clients/compute/__init__.py

```diff
@@ -441,3 +441,56 @@ def delete_floating_ip(self, tenant_id, fip_id=None):
         """
         r = self.floating_ips_delete(tenant_id, fip_id)
         return r.headers
+
+    def list_keypairs(self):
+        """
+        :returns: (list) [
+            {id: ..., name: ..., public_key: ..., fingerprint: ...},
+            ... ]
+        """
+        r = self.keypairs_get()
+        return r.json['keypairs']
+
+    def get_keypair_details(self, key_name=None):
+        """
+        :param key_name: (str) The name of the key
+
+        :returns: (dict) {keypair: ..., id: ..., name: ..., public_key: ...,
+                                   fingerprint: ...., created_at: ...,
+                                   updated_at: ...}
+        """
+        r = self.keypairs_get(key_name=key_name)
+        return r.json['keypair']
+
+    def create_or_update_key(self, key_name=None, public_key=None):
```

Comment:
Well, a key_name is just a string. It does not have any complexity, so I don't think there is a need to try to release the library user from the burden of creating one. If you do implement the auto generation here, then you will probably have to give the user the ability to create a name according to a user defined template too, which makes this feature an "overkill" in the first place. It's better if you force the user to provide a keyname.